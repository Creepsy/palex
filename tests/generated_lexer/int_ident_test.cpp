#include <string_view>
#include <iostream>

#include <IntIdentTestLexer.h>

int main() {
    const std::string_view input = "1000 anIdent another identifier 2000 \n 1999 number 99";
    palex::IntIdentTestLexer lexer(input);
    do {
        lexer.next_unignored_token();
        std::cout << lexer.current_token() << std::endl;
    } while (lexer.current_token().type != palex::IntIdentTestToken::TokenType::END_OF_FILE);
    return 0;
}