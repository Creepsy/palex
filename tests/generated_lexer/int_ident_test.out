[Ln 1, Col 1] - [Ln 1, Col 5] INT: 1000
[Ln 1, Col 6] - [Ln 1, Col 13] IDENTIFIER: anIdent
[Ln 1, Col 14] - [Ln 1, Col 21] IDENTIFIER: another
[Ln 1, Col 22] - [Ln 1, Col 32] IDENTIFIER: identifier
[Ln 1, Col 33] - [Ln 1, Col 37] INT: 2000
[Ln 2, Col 2] - [Ln 2, Col 6] INT: 1999
[Ln 2, Col 7] - [Ln 2, Col 13] IDENTIFIER: number
[Ln 2, Col 14] - [Ln 2, Col 16] INT: 99
[Ln 2, Col 16] - [Ln 2, Col 16] END_OF_FILE: 
